docs/RFCS/sql_privileges.md

```diff
@@ -0,0 +1,208 @@
+- Feature Name: SQL privileges
+- Status: draft
+- Start Date: 2015-08-10
+- RFC PR: [#2054](https://github.com/cockroachdb/cockroach/pull/2054)
+- Cockroach Issue: [#2005](https://github.com/cockroachdb/cockroach/issues/2005)
+
+# Summary
+
+Introduce SQL privileges for databases and tables. The aim is to provide a privilege
+framework familiar to users of popular SQL servers while embracing simplicity.
+
+We intentionally do not address privileges for other granularities (columns or rows),
+or unsupported features (views, temporary tables, procedures, etc...).
+
+# Motivation
+
+The SQL interface needs permissioning on database and tables.
+For ease of migration and understanding, we wish to use common SQL privileges
+albeit adapted for our use, or simplified.
+
+We use the [postgres](http://www.postgresql.org/docs/9.4/static/sql-grant.html) and
+[mysql](https://dev.mysql.com/doc/refman/5.7/en/privileges-provided.html) privileges
+as references.
+
+# Detailed design
+
+## Privilege logic
+
+### Postgres
+
+* all objects have owners. Owners default to all privileges on their objects.
+* only the granter of a privilege can remove it. (eg: user A grants `SELECT` to B, only
+  A can remove `SELECT` from B. If another role granted `SELECT` as well, it stays).
+* databases share no data between them except for user tables. As such they are closer
+  to our databases then the postgres schema.
+* `SUPERUSER` is a user attribute. It allows everything.
+* postgres has a `PUBLIC` schema granting liberal privileges to all users.
+* privileges can be displayed using the `\du` command in psql.
+* `ALL` is translated to the list of all privileges.
+
+### Mysql
+
+* privileges are stored in a global table. It is thus possible to grant privileges
+  on objects that do not yet exist (eg: `GRANT ALL on db.* TO user1`).
+* privileges can be displayed using `SHOW GRANTS [FOR x]`.
+* `ALL` is a shortcut for setting and display.
+
+### Cockroach
+
+* the `root` user has default `ALL` privileges on all new databases. It is also the only
+  user allowed to create databases.
+* `root` user privileges cannot be removed. (This may change and be implicit `ALL` for `root`).
+* table privileges are inherited from database privileges at creation time. After that,
+  they are allowed to diverge.
+* `GRANT` is specified as a privilege. It may make more sense to use `WITH GRANT OPTION`
+  on the `GRANT` statement (similar to postgres and mysql).
+* privileges can be displayed using `SHOW GRANTS [FOR x]`.
+* `ALL` is a shortcut for setting and display.
+
+## Supported privileges
+
+| Privilege | Level     |
+|-----------|-----------|
+| ALL       | DB, Table |
+| CREATE    | DB        |
```

Comment:
this is the level at which the permission is stored, not what they're used for.
`CREATE TABLE` is documented further down, it requires `CREATE` at the database level.
`CREATE DATABASE` requires the `root` user.